### PR TITLE
update: change components captions structure

### LINKED ISSUE #1253: Altering margin/border on Input element using style props not working as expected
Edit: Made a few edits as I found a hacky workaround.

Hi I am running UI Kitten 4.x and I am trying to eliminate the border and margin from the bottom of an input field.

Currently this is the only approach that "works":

```
{
   marginBottom: -4,
   borderBottomColor: 'transparent',
   borderBottomLeftRadius: 0,
   borderBottomRightRadius: 0
}
```

This feels somewhat hacky. 1) I am not totally sure how good of an idea it is to go with negative margins, and 2) I am not truly eliminating the border but just making it invisible. `borderBottomWidth: 0` seems like it would be preferable as well as `marginBottom: 0`. Is there a less hacky way to make this work?

Screen Shot 2020-09-28 at 4 30 07 PM

### PR DESCRIPTION
**Pull request by whitestranger7**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:
Closes #1253
[Related eva fix.](https://github.com/eva-design/eva/pull/75)
[Related kittenTricks update.](https://github.com/akveo/kittenTricks/pull/319)

### CODE REVIEW

**Review by artyorsh — COMMENTED** [review]
This will break the final look of the component configured with `caption` or `captionIcon` props. Since we don't use this default spacing between caption text and Input field, a user will more likely have to apply it on his own. Which makes Eva not Eva.

Instead, I would suggest removing the `captionIcon` property and wrap the entire `caption` element in `FalsyFC` or better `FalsyText` so we can remove the [caption container view](https://github.com/akveo/react-native-ui-kitten/blob/1f2ed739eabb696e4e0cb7c7eb915142e122167e/src/components/ui/input/input.component.tsx#L302). This way of composing elements will allow us reducing useless spacings between components.

**Comment by whitestranger7** [comment]
> This will break the final look of the component configured with `caption` or `captionIcon` props. Since we don't use this default spacing between caption text and Input field, a user will more likely have to apply it on his own. Which makes Eva not Eva.

Im not really sure that I got your point, because I did some tests and this additional `marginTop: 4px` doesn't affect component with `caption` and `captionIcon` props.
Screenshot 2021-02-18 at 17 12 49

File: src/components/ui/input/input.component.tsx [hunk 1]
```diff
@@ -292,9 +292,11 @@ export class Input extends React.Component<InputProps> implements WebEventRespon
...
           />
         </View>
-        <FalsyFC
-          component={caption} 
-        />
+        {typeof caption === "function" ? (
+          <FalsyFC style={evaStyle.captionLabel} component={caption} />
+        ) : (
+          <FalsyText style={evaStyle.captionLabel} component={caption} />
+        )}
```
Comment (artyorsh) [inline src/components/ui/input/input.component.tsx:299]: FalsyText will work in both cases

**Review by artyorsh — CHANGES_REQUESTED** [review]

File: src/components/ui/input/input.component.tsx [hunk 2]
```diff
@@ -302,16 +292,7 @@ export class Input extends React.Component<InputProps> implements WebEventRespon
...
-          <FalsyFC
-            style={evaStyle.captionIcon}
-            component={captionIcon}
-          />
-          <FalsyText
-            style={[evaStyle.captionLabel, styles.captionLabel]}
-            component={caption}
-          />
-        </View>
+        <FalsyText style={evaStyle.captionLabel} component={caption} />
```
Comment (malashkevich) [inline src/components/ui/input/input.component.tsx:295]: multiline

File: src/components/ui/input/input.spec.tsx [hunk 3]
```diff
@@ -121,21 +128,21 @@ describe('@input: component checks', () => {
   });
 
   it('should render component passed to captionIcon prop', () => {
```
Comment (malashkevich) [inline src/components/ui/input/input.spec.tsx:130]: incorrect name

**Review by artyorsh — APPROVED** [review]

**Pull request merged**